sh/drive-utils.sh

```diff
@@ -0,0 +1,535 @@
+    [ -n "${id_check_existing_file}" ] && printf "%s\n" "${search_response_check_existing_file}"
+    return 0
+)
+
+###################################################
+# Create/Check directory in google drive.
+# Globals: 2 variables, 2 functions
+#   Variables - API_URL, API_VERSION
+#   Functions - _url_encode, _json_value
+# Arguments: 3
+#   ${1} = dir name
+#   ${2} = root dir id of given dir
+#   ${3} = Access Token
+# Result: print folder id
+# Reference:
+#   https://developers.google.com/drive/api/v3/folder
+###################################################
+_create_directory() (
+    [ $# -lt 3 ] && printf "Missing arguments\n" && return 1
+    dirname_create_directory="${1##*/}" rootdir_create_directory="${2}" token_create_directory="${3}"
+    unset query_create_directory search_response_create_directory folder_id_create_directory
+
+    "${EXTRA_LOG:-:}" "justify" "Creating gdrive dir.." "-" 1>&2
+    query_create_directory="$(_url_encode "mimeType='application/vnd.google-apps.folder' and name='${dirname_create_directory}' and trashed=false and '${rootdir_create_directory}' in parents")"
+
+    search_response_create_directory="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
+        -H "Authorization: Bearer ${token_create_directory}" \
+        "${API_URL}/drive/${API_VERSION}/files?q=${query_create_directory}&fields=files(id)&supportsAllDrives=true" || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+
+    folder_id_create_directory="$(printf "%s\n" "${search_response_create_directory}" | _json_value id 1 1)"
+
+    if [ -z "${folder_id_create_directory}" ]; then
+        unset create_folder_post_data_create_directory create_folder_response_create_directory
+        create_folder_post_data_create_directory="{\"mimeType\": \"application/vnd.google-apps.folder\",\"name\": \"${dirname_create_directory}\",\"parents\": [\"${rootdir_create_directory}\"]}"
+        create_folder_response_create_directory="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
+            -X POST \
+            -H "Authorization: Bearer ${token_create_directory}" \
+            -H "Content-Type: application/json; charset=UTF-8" \
+            -d "${create_folder_post_data_create_directory}" \
+            "${API_URL}/drive/${API_VERSION}/files?fields=id&supportsAllDrives=true" || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+        folder_id_create_directory="$(printf "%s\n" "${create_folder_response_create_directory}" | _json_value id 1 1)"
+    fi
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    { [ -z "${folder_id_create_directory}" ] && printf "%s\n" "${create_folder_response_create_directory}" 1>&2 && return 1; } || {
+        printf "%s\n" "${folder_id_create_directory}"
+    }
+    return 0
+)
+
+###################################################
+# Sub functions for _upload_file function - Start
+# generate resumable upload link
+_generate_upload_link() {
+    uploadlink_upload_file="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)